src/plugins/auth/auth.ts

```diff
@@ -329,6 +332,61 @@ const plugin: FastifyPluginAsync<AuthPluginOptions> = async (fastify, options) =
       },
     );
 
+    async function verifyCurrentPassword(member: Member, password: string) {
+      /* verified: stores the output of bcrypt.compare().
+      bcrypt.compare() allows to compare the provided password with a stored hash. 
+      It deduces the salt from the hash and is able to then hash the provided password correctly for comparison
+      if they match, verified is true 
+      if they do not match, verified is false
+      */
+      // if the member already has a password set: return verified
+      if (member.password) {
+        const verified = bcrypt
+          .compare(password, member.password)
+          .then((res) => res)
+          .catch((err) => console.error(err.message));
+        return verified;
+      }
+      // if the member does not have a password set: return true
+      return true;
+    }
+
+    async function encryptPassword(password: string) {
+      /* encrypted: stores the output of bcrypt.hash().
+      bcrypt.hash() creates the salt and hashes the password. 
+      A new hash is created each time the function is run, regardless of the password being the same. 
+      */
+      const saltRounds = 10;
+      const encrypted = bcrypt
+        .hash(password, saltRounds)
+        .then((hash) => hash)
+        .catch((err) => console.error(err.message));
+      return encrypted;
+    }
+
+    // update member password
+    fastify.patch<{ Params: IdParam }>(
+      '/members/:id/update-password',
+      { schema: updatePassword },
+      async ({ params: { id }, body, log }) => {
+        const member = await mS.get(id, db.pool);
+        // verify that input current password is the same as the stored one
```

Comment:
Actually you never check that actor is the member. But actually we should go for the following endpoint `/members/update-password` and assume it's for the current member. We will have another endpoint for admin where we can precise the member id.